custom/icds/translations/integrations/client.py

```diff
@@ -66,3 +69,31 @@ def project_details(self):
         return requests.get(
             url, auth=self._auth,
         )
+
+    def resource_details(self, resource_slug, lang):
+        url = "https://www.transifex.com/api/2/project/{}/resource/{}/stats/{}".format(
+            self.project, resource_slug, lang)
+        response = requests.get(url, auth=self._auth)
+        if response.status_code == 200:
+            return response.json()
+        elif response.status_code == 404:
+            raise ResourceMissing("Resource {} not found for lang {}".format(resource_slug, lang))
+        raise Exception(response.content)
+
+    def confirm_complete_translation(self, resource_slug, lang):
+        """
+        check if a resource has been completely translated for the target lang
+        """
+        return self.resource_details(resource_slug, lang)['completed'] == "100%"
+
+    def get_translation(self, resource_slug, lang):
+        url = "https://www.transifex.com/api/2/project/{}/resource/{}/translation/{}/?file".format(
+            self.project, resource_slug, lang
+        )
+        response = requests.get(url, auth=self._auth, stream=True)
+        if response.status_code != 200:
+            return response
+        temp_file = tempfile.NamedTemporaryFile()
```

Comment:
Just to confirm
```
temp_file = tempfile.NamedTemporaryFile()
temp_file.write(response.content)
temp_file.seek(0)
polib_obj = polib.pofile(temp_file.name)
temp_file.close()
```
This looks good?

Reply:
Actually I don't think you'll need the `seek` line.  Since you're not passing the open file reference to `polib`, it must open the file again.  Otherwise that looks correct.

You'll need to make sure `tempfile.close()` doesn't get closed until `polib_obj` is no longer needed, since it'll delete the file.  Similarly, it's safer to put all operations on this temfile inside a `with` statement so the file gets cleaned up even on failure.  That can look like this:
```python
with tempfile.NamedTemporaryFile() as tempfile:
    temp_file.write(response.content)
    polib_obj = polib.pofile(temp_file.name)
```
The downside of this is that `polib_obj` maintains a separate handler to the same file, so you'll have the file open twice.  I believe having a file open multiple times works on most systems (just not Windows, I think?).  It could get confusing if both tried writing to it in sequence.  That `TransientTempfile` util I mentioned in my review comment lets you close the file without yet deleting it.  This use-case is probably simple enough that it won't run into issues either way, though.

Reply:
> Actually I don't think you'll need the seek line.

I tried this and just using `write` didnt write to file, had to do `seek` to make it actually write the content.

> You'll need to make sure tempfile.close() doesn't get closed until polib_obj is no longer needed, since it'll delete the file

Ya i think we don't need that file after that. `polib.pofile(temp_file.name)` parses the file and converts it to `POEntries`. i have updated this now and there is no `po_obj`.

Reply:
> I tried this and just using write didnt write to file, had to do seek to make it actually write the content.

`seek(0)` moves the pointer back to the beginning of the file.  `write` does write to the file, but you can't read from the same file object again without resetting the pointer.  `polib.pofile` only gets the file name, so it must be opening the file a second time, which means it *should* be able to read without seeking.  Here's an example:
```python
In [1]: import tempfile

In [2]: with tempfile.NamedTemporaryFile() as f:
   ...:     f.write('hello')
   ...:     print "read 1", f.read()
   ...:     f.seek(0)
   ...:     print "read 2", f.read()
   ...:     print "read 3", f.read()
   ...:     with open(f.name) as f2:
   ...:         print "read 4", f2.read()
   ...:         
read 1 
read 2 hello
read 3 
read 4 hello
```

> Ya i think we don't need that file after that. polib.pofile(temp_file.name) parses the file and converts it to POEntries. i have updated this now and there is no po_obj.

Cool yeah, as long as that doesn't need the tempfile after initialization this should be fine.  I wasn't sure whether it streamed the file lazily or loaded it all into memory up front, for example.

Reply:
@esoergel tried making this change and ran into some issue with encoding because of it.
Want to leave it as-is for now and look at this improvement later.

Also looks like with the current implementation,
it should be safe to leave as-is. When the `temp_file` reference is removed, the file will get removed as well, so there should be no orphan file references, temp files even if anything goes wrong. let me know if thats okay.